src/utils/blocks.js

```diff
@@ -66,16 +66,160 @@ export function getBlockIcon( item ) {
 }
 
 /**
- * Serializes inserter items to a format suitable for native consumption.
- * Extracts only the properties needed by the native side and ensures
- * proper formatting (e.g., converting React icon elements to SVG strings).
+ * Predefined category ordering.
+ * Display names will be retrieved from WordPress categories for proper localization.
+ */
+const ORDERED_CATEGORIES = [
+	{ key: 'text' },
+	{ key: 'media' },
+	{ key: 'design' },
+	{ key: 'widgets' },
+	{ key: 'theme' },
+	{ key: 'embed' },
+];
+
+/**
+ * Predefined block ordering within categories.
+ */
+const BLOCK_ORDER_BY_CATEGORY = {
+	text: [
+		'core/paragraph',
+		'core/heading',
+		'core/list',
+		'core/list-item',
+		'core/quote',
+		'core/code',
+		'core/preformatted',
+		'core/verse',
+		'core/table',
+	],
+	media: [
+		'core/image',
+		'core/video',
+		'core/gallery',
+		'core/embed',
+		'core/audio',
+		'core/file',
+	],
+	design: [
+		'core/separator',
+		'core/spacer',
+		'core/columns',
+		'core/column',
+	],
+	embed: [
+		'core/embed', // Generic embed - always first
+		'core/embed/youtube',
+		'core/embed/vimeo',
+		'core/embed/tiktok',
+		'core/embed/wordpress',
+		'core/embed/tumblr',
+		'core/embed/videopress',
+		'core/embed/pocket-casts',
+		'core/embed/reddit',
+		'core/embed/pinterest',
+		'core/embed/spotify',
+		'core/embed/soundcloud',
+	],
+};
+
+/**
+ * Most used blocks to show when there are no contextual blocks.
+ * Optimized for mobile usage patterns where users frequently create content
+ * with text, media from device cameras, and visual formatting.
+ */
+const MOST_USED_BLOCKS = [
```

Comment:
Feedback on `MOST_USED_BLOCKS` and `BLOCK_ORDER_BY_CATEGORY` is welcome.
I expect to refine it further in the scope of [CMM-915: Add first-class support for Jetpack blocks](https://linear.app/a8c/issue/CMM-915/add-first-class-support-for-jetpack-blocks).